test/js/test.js

```diff
@@ -348,18 +349,59 @@ function getReadyToValidate(err, resp, body) {
   var authz = JSON.parse(body);
 
   var httpChallenges = authz.challenges.filter(function(x) { return x.type == "http-01"; });
-  if (httpChallenges.length == 0) {
+  var dnsChallenges = authz.challenges.filter(function(x) { return x.type == "dns-01"; });
```

Comment:
You can simplify here by doing

`var challenges = authz.challenges.filter(function(x) { return x.type == cliOptions.challType; });`